src/Views/SilenceForm/Views.elm

```diff
@@ -1,74 +1,36 @@
-module Views.SilenceForm.Views exposing (edit, new)
+module Views.SilenceForm.Views exposing (view)
 
-import Html exposing (Html, div, text, fieldset, legend, label, span, a)
+import Html exposing (Html, a, div, fieldset, label, legend, span, text)
 import Html.Attributes exposing (class, href)
 import Html.Events exposing (onClick)
-import Silences.Types exposing (Silence)
-import Types exposing (Msg(MsgForSilenceList, PreviewSilence, MsgForSilenceForm), Model)
-import Utils.Types exposing (Matcher, ApiResponse(Success, Loading, Failure), ApiData)
-import Utils.Views exposing (loading, error, checkbox)
+import Silences.Types exposing (Silence, SilenceId)
+import Utils.Types exposing (ApiData, ApiResponse(..), Matcher)
+import Utils.Views exposing (checkbox, error, formField, formInput, iconButtonMsg, textField)
 import Views.Shared.SilencePreview
-import Utils.Views exposing (formField, iconButtonMsg, textField, formInput)
-import Views.SilenceForm.Types
-    exposing
-        ( SilenceFormMsg(UpdateStartsAt, UpdateCreatedBy, UpdateDuration, UpdateEndsAt, UpdateComment, AddMatcher, CreateSilence, DeleteMatcher, UpdateMatcherRegex, UpdateMatcherValue, UpdateMatcherName)
-        )
-
-
-edit : ApiData Silence -> Html Msg
-edit silence =
-    case silence of
-        Success silence ->
-            silenceForm "Edit" silence
-
-        Loading ->
-            loading
-
-        Failure msg ->
-            error msg
-
-
-new : ApiData Silence -> Html Msg
-new silence =
-    case silence of
-        Success silence ->
-            silenceForm "New" silence
-
-        Loading ->
-            loading
-
-        Failure msg ->
-            error msg
+import Views.SilenceForm.Types exposing (Model, SilenceFormMsg(..))
+import Utils.Views exposing (checkbox, formField, formInput, iconButtonMsg, textField)
+import Views.Shared.SilencePreview
+import Views.SilenceForm.Types exposing (Model, SilenceFormMsg(..), SilenceFormFieldMsg(..))
 
 
-silenceForm : String -> Silence -> Html Msg
-silenceForm kind silence =
-    -- TODO: Add field validations.
+view : Maybe SilenceId -> Model -> Html SilenceFormMsg
+view maybeId { silence, form } =
```

Comment:
I think it only makes sense to separate two forms when these two cases diverge, as @stuartnelson3 pointed below, there is no editing, it always creates a new silence.